compose/service/locale.go

```diff
@@ -272,6 +272,43 @@ func (svc resourceTranslationsManager) pageExtended(ctx context.Context, res *ty
 	return
 }
 
+func (svc resourceTranslationsManager) chartExtended(_ context.Context, res *types.Chart) (out locale.ResourceTranslationSet, err error) {
+	var (
+		yAxisLabelK  = types.LocaleKeyChartYAxisLabel
+		metricLabelK = types.LocaleKeyChartMetricsMetricIDLabel
+	)
+
+	for _, report := range res.Config.Reports {
+		for _, tag := range svc.locale.Tags() {
+			out = append(out, &locale.ResourceTranslation{
+				Resource: res.ResourceTranslation(),
+				Lang:     tag.String(),
+				Key:      yAxisLabelK.Path,
+				Msg:      svc.locale.TResourceFor(tag, res.ResourceTranslation(), yAxisLabelK.Path),
+			})
+		}
+
+		for _, metric := range report.Metrics {
+			if _, ok := metric["metricID"]; ok {
+				mpl := strings.NewReplacer(
+					"{{metricID}}", metric["metricID"].(string),
```

Comment:
This feels a bit dangerous to me; if it's not ok it will panic
I see you do some validation before this, but I would still prefer you do the *more safe* type assertion.
Alternatively, you can add a bunch of tests to assure this will be fine.

Reply:
Apply same comment for the rest of such cases